src/ui/profile_preferences.ui

```diff
@@ -1916,6 +2059,192 @@ you can use it but there could be issues with aligning columns of text</string>
             </property>
            </widget>
           </item>
+          <item row="14" column="0" colspan="4">
+           <widget class="QCheckBox" name="checkBox_boldIsBright">
+            <property name="toolTip">
+             <string>&lt;p&gt;Server uses BOLD to make Light 8 colors (only for some 16 Color MUDs)&lt;/p&gt;&lt;p&gt;Some older Game Servers support eight different colors which are selected by using the original eight &quot;Set Graphic Rendition&quot; &lt;tt&gt; &amp;lt;SGR&amp;gt; 30&lt;/tt&gt; (black) to &lt;tt&gt;37&lt;/tt&gt; (white)&lt;/tt&gt; ANSI ESC color codes and combine those with the &lt;tt&gt;&amp;lt;SGR&amp;gt; 1&lt;/tt&gt; (bold or more intense) code to provide a second set of eight colors for sixteen in total. This checkbox causes this use of the &lt;tt&gt;&amp;lt;SGR&amp;gt; 1&lt;/tt&gt; code however it should not be used if the Game Server:&lt;ul&gt;&lt;li&gt;only uses eight colors&lt;/li&gt;&lt;li&gt;uses more than sixteen colors&lt;/li&gt;&lt;li&gt;uses sixteen colors but selects the &quot;brighter&quot; eight colors with &lt;tt&gt;&amp;lt;SGR&amp;gt; 90&lt;/tt&gt; (light black) to &lt;tt&gt;&amp;lt;SGR&amp;gt; 97&lt;/tt&gt; (light white) {and &lt;tt&gt;100&lt;/tt&gt; to &lt;tt&gt;107&lt;/tt&gt; for the same as a background} of the sixteeen&lt;/li&gt;&lt;/ul&gt;&lt;/p&gt;&lt;p&gt;&lt;i&gt;Older versions of Mudlet had code that effectively had this option permanently enabled. However nowadays it interferes with using the BOLD attribute to provide &lt;b&gt;bold&lt;/b&gt; text alongside the less often used FAINT attribute to provide &lt;span style=&quot; font-weight: 200&quot;&gt;faint&lt;/span&gt; and &lt;span style=&quot; font-weight: 600&quot;&gt;demi-bold&lt;/span&gt; text effects.&lt;/p&gt;</string>
+            </property>
+            <property name="text">
+             <string>BOLD is Bright</string>
+            </property>
+           </widget>
+          </item>
+         </layout>
+        </widget>
+       </item>
+       <item>
+        <widget class="QGroupBox" name="groupBox_sgrCodesSeen">
+         <property name="title">
+          <string>&quot;Set Graphics Rendition&quot; codes detected (check this to show details)</string>
```

Comment:
The settings' menu purpose for working with settings and I'm not certain diluting it to do debugging as well is a good idea.

Reply:
It is the most appropriate place that I can think of to implement it - if you can see from that that you are only getting 8 ANSI colours - like you do from, say, WotMUD, then you can work out that you probably want (like I do) to swap the first set of 8 colours with the second 8. Also, if you can see that your MUD is sending you AXTerm colour codes to do the second half of the 16 Colour set then you can reasonable uncheck the "BOLD-is-Bright" option) - as you can if it is sending 256 or 16M colour codes. Anyhow I think it will help us and others debug problems when end-users complain about their MUD not showing the colours correctly.

Knowing that you are seeing Alternate font/blinking/concealed codes that Mudlet does not currently render will help us and MUD Server operators develop these features - should we/they wish to in the future.

Reply:
I think this is addressing a problem that we think exits in theory but doesn't actually happen in practice. I haven't seen players struggle with this, but maybe I missed something? 

This makes the case for diluting what settings is about not very strong 😕

At most, this can perhaps be worked into the text analyser some way - even though the text analyser looks at one character at a time and this tool looks at all the text that as been received from the last reset.

Reply:
Okay - under protest - I've whipped out all the extra bits try 0611553b0928b8f38ddbc49dd9c472b4f8bb7d95 now... :cry: